src/OffsetArrays.jl

```diff
@@ -370,6 +373,20 @@ end
 
 no_offset_view(A::OffsetArray) = no_offset_view(parent(A))
 
+# Quick hack for matrix multiplication.
+# Ideally, one would instead improve LinearAlgebra's support of custom indexing.
+function Base.:(*)(A::OffsetMatrix, B::OffsetMatrix)
+    matmult_check_axes(A, B)
+    C = OffsetArray(parent(A) * parent(B), (axes(A,1), axes(B,2)))
+end
```

Comment:
```suggestion
function Base.:(*)(A::OffsetMatrix, B::OffsetMatrix)
    matmult_check_axes(A, B)
    C = parent(A) * parent(B)
    OffsetArray{eltype(C), 2, typeof(C)}(C, (A.offsets[1], B.offsets[2]))
end
```

Better to avoid usage of `axes` here because that involves unnecessary offset computation via `IdOffsetRange`.

And similar to the matrix-times-vector case.

Reply:
I've implemented this in the latest commit